core/src/components/modal/gestures/utils.ts

```diff
@@ -18,7 +18,7 @@ export const handleCanDismiss = async (el: HTMLIonModalElement, animation: Anima
    * If the function returns `true`,
    * then we can proceed with dismiss.
    */
-  const shouldDismiss = await el.canDismiss();
+  const shouldDismiss = await el.canDismiss(undefined, 'gesture');
```

Comment:
~ For backdrop we have an explicit constant to track the role (https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/overlays.ts#L614). Should we do something similar here? 

Alternatively, how would we feel about an enum/const for either `ModalRoles` or `OverlayRoles` in the future vs. developers needing to know which explicit constant or string value the role is.

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/26384/commits/d21111d9b1d0739d43e904bcb61da3f5871f263a.